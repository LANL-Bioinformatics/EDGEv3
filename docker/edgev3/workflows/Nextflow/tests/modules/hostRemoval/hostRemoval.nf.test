nextflow_workflow {

    name "Test Workflow HOSTREMOVAL"
    script "modules/hostRemoval/hostRemoval.nf"
    workflow "HOSTREMOVAL"

    test("PE Reads, Single Host") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testHR"
                }
                hostRemoval {
                    host = ["${projectDir}/test_data/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz"]
                    cpus = 4
                }
                
            }
            workflow {
                """
                input[0] = params.hostRemoval.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                """
            }
        }

        then {
            
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.paired[0].find { file(it).name == "hostclean.1.fastq" }).exists()},
                {assert file(workflow.out.paired[0].find { file(it).name == "hostclean.2.fastq" }).exists()},
                {assert file(workflow.out.unpaired.find { file(it).name == "hostclean.unpaired.fastq" }).exists()}
            )
        }

    }

    test("SE Reads, Single Host") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testHR"
                }
                hostRemoval {
                    host = "${projectDir}/test_data/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz"
                    cpus = 4
                }
                
            }
            workflow {
                """
                input[0] = params.hostRemoval.plus(params.shared)
                input[1] = [file("${projectDir}/nf_assets/NO_FILE")]
                input[2] = file("${projectDir}/test_data/Ecoli_10x.1.fastq")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.unpaired.find { file(it).name == "hostclean.unpaired.fastq" }).exists()},
            )
        }

    }

    test("SE Reads, Multi Host") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testHR"
                }
                hostRemoval {
                    host = ["${projectDir}/test_data/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz", "${projectDir}/test_data/Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz"]
                    cpus = 4
                }
            }
            workflow {
                """
                input[0] = params.hostRemoval.plus(params.shared)
                input[1] = [file("${projectDir}/nf_assets/NO_FILE")]
                input[2] = file("${projectDir}/test_data/Ecoli_10x.1.fastq")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.unpaired.find { file(it).name == "hostclean.unpaired.fastq" }).exists()},
            )

        }

    }

    test("PE Reads, Multi Host") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testHR"
                }
                hostRemoval {
                    host = ["${projectDir}/test_data/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz","${projectDir}/test_data/Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz"]
                    cpus = 4
                }
            }
            workflow {
                """
                input[0] = params.hostRemoval.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.paired[0].find { file(it).name == "hostclean.1.fastq" }).exists()},
                {assert file(workflow.out.paired[0].find { file(it).name == "hostclean.2.fastq" }).exists()},
                {assert file(workflow.out.unpaired.find { file(it).name == "hostclean.unpaired.fastq" }).exists()},
            )
              
        }

    }


}
