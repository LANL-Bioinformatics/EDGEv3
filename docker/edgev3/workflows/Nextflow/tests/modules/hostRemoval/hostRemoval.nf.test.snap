{
    "SE Reads, Single Host": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "hostclean.unpaired.fastq:md5,7b1d185d8e9fef1fc68efba3257ca3fb"
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    "hostclean.unpaired.fastq:md5,7b1d185d8e9fef1fc68efba3257ca3fb"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-20T18:45:45.584906"
    },
    "PE Reads, Multi Host": {
        "content": [
            {
                "0": [
                    [
                        "hostclean.1.fastq:md5,e2f29a1147a9f450557cf88ad72400cf",
                        "hostclean.2.fastq:md5,70927fcae9ee7bdbd98fce66b5d353f3"
                    ]
                ],
                "1": [
                    "hostclean.unpaired.fastq:md5,a1ace46fc8b5a225afcab35e757e0542"
                ],
                "paired": [
                    [
                        "hostclean.1.fastq:md5,e2f29a1147a9f450557cf88ad72400cf",
                        "hostclean.2.fastq:md5,70927fcae9ee7bdbd98fce66b5d353f3"
                    ]
                ],
                "unpaired": [
                    "hostclean.unpaired.fastq:md5,a1ace46fc8b5a225afcab35e757e0542"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-20T18:35:06.56628"
    },
    "PE Reads, Single Host": {
        "content": [
            {
                "0": [
                    [
                        "hostclean.1.fastq:md5,de55918c54218d510baf01946e97e28b",
                        "hostclean.2.fastq:md5,6395b18db7f09a6374495d577533dfc0"
                    ]
                ],
                "1": [
                    "hostclean.unpaired.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "paired": [
                    [
                        "hostclean.1.fastq:md5,de55918c54218d510baf01946e97e28b",
                        "hostclean.2.fastq:md5,6395b18db7f09a6374495d577533dfc0"
                    ]
                ],
                "unpaired": [
                    "hostclean.unpaired.fastq:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-20T18:41:15.089008"
    },
    "SE Reads, Multi Host": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "hostclean.unpaired.fastq:md5,8fc35373e3bfb3951e9d7632f3362861"
                ],
                "paired": [
                    
                ],
                "unpaired": [
                    "hostclean.unpaired.fastq:md5,8fc35373e3bfb3951e9d7632f3362861"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-20T18:27:35.91726"
    }
}